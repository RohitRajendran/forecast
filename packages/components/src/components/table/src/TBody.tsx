import React from "react";
import { Box } from "../../..";

type TBodyProps = {
  children?: React.ReactNode;
};

/** Used to group the body content in a table */
const TBody = React.forwardRef<HTMLTableSectionElement, TBodyProps>(
  ({ children }, ref) => {
    return (
      <Box ref={ref} as="tbody">
        {children}
      </Box>
    );
  }
);

TBody.displayName = "TBody";

export { TBody };
