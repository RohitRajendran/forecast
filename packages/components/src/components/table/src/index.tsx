export * from "./Table";
export * from "./TBody";
export * from "./TD";
export * from "./TH";
export * from "./THead";
export * from "./TR";
