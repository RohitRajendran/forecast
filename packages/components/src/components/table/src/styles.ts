import { styled, theme } from "@trueplan/forecast-theme";

const border = `1px solid ${theme.colors.gray40}`;
const cellPadding = `$30 $35`;

export const StyledTable = styled("table", {
  border,
  backgroundColor: theme.colors.white,
  borderCollapse: "collapse",
  width: "100%",
  variants: {
    tableLayout: {
      auto: {
        tableLayout: "auto",
      },
      fixed: {
        tableLayout: "fixed",
      },
    },
  },
});

export const StyledTHead = styled("thead", {
  borderBottom: border,
});

export const StyledTR = styled("tr", { borderBottom: border });

export const StyledTD = styled("td", {
  padding: cellPadding,
});

export const StyledTH = styled("th", {
  padding: cellPadding,
  textAlign: "left",
});
