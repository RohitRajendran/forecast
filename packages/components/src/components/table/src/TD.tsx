import React from "react";
import { StyledTD } from "./styles";

type TDProps = {
  children?: React.ReactNode;
  /** To make a cell span over multiple rows. */
  rowSpan?: number;
  /** To make a cell span over multiple columns. */
  colSpan?: number;
};

/** A standard cell in the table, use within a <TR>. */
const TD = React.forwardRef<HTMLTableCellElement, TDProps>(
  ({ children, rowSpan, colSpan }, ref) => {
    return (
      <StyledTD ref={ref} rowSpan={rowSpan} colSpan={colSpan}>
        {children}
      </StyledTD>
    );
  }
);

TD.displayName = "TD";

export { TD };
