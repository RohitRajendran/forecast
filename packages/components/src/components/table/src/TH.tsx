import React from "react";
import { StyledTH } from "./styles";

type THProps = {
  children?: React.ReactNode;
  /** To make a cell span over multiple rows. */
  rowSpan?: number;
  /** To make a cell span over multiple columns */
  colSpan?: number;
};

/** A header cell in the table, use within a <THead>. */
const TH = React.forwardRef<HTMLTableCellElement, THProps>(
  ({ children, rowSpan, colSpan }, ref) => {
    return (
      <StyledTH ref={ref} rowSpan={rowSpan} colSpan={colSpan}>
        {children}
      </StyledTH>
    );
  }
);

TH.displayName = "TH";

export { TH };
