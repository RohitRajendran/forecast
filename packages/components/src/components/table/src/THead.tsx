import React from "react";
import { StyledTHead } from "./styles";

type THeadProps = {
  children?: React.ReactNode;
};

/** Used to group header content in a table */
const THead = React.forwardRef<HTMLTableSectionElement, THeadProps>(
  ({ children }, ref) => {
    return <StyledTHead ref={ref}>{children}</StyledTHead>;
  }
);

THead.displayName = "THead";

export { THead };
