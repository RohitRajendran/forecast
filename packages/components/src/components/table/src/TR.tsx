import React from "react";
import { StyledTR } from "./styles";

type TRProps = {
  children?: React.ReactNode;
};

/** A row in the table */
const TR = React.forwardRef<HTMLTableRowElement, TRProps>(
  ({ children }, ref) => {
    return <StyledTR ref={ref}>{children}</StyledTR>;
  }
);

TR.displayName = "TR";

export { TR };
