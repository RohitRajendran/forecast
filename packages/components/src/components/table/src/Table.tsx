import React from "react";
import { StyledTable } from "./styles";

type TableProps = {
  /** Determines how to layout the cells, rows, and columns=*/
  tableLayout?: "fixed" | "auto";
  children?: React.ReactNode;
};

/** Represent your data in rows and columns */
const Table = React.forwardRef<HTMLTableElement, TableProps>(
  ({ children, tableLayout = "auto" }, ref) => {
    return (
      <StyledTable ref={ref} tableLayout={tableLayout}>
        {children}
      </StyledTable>
    );
  }
);

Table.displayName = "Table";

export { Table };
