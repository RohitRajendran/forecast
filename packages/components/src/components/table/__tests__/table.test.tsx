import { render } from "@testing-library/react";
import React from "react";
import { Table, TBody, TD, TH, THead, TR } from "../src";

describe("Table", () => {
  it("should render", async () => {
    const { container } = render(
      <Table tableLayout="fixed">
        <THead>
          <TH>File Name</TH>
          <TH>Date Uploaded</TH>
          <TH>Imported By</TH>
          <TH>Import Status</TH>
        </THead>
        <TBody>
          <TR>
            <TD>trueplan2.csv</TD>
            <TD>May 24, 2022 at 3:23pm</TD>
            <TD>Olivia Smith</TD>
            <TD>Failed</TD>
          </TR>
        </TBody>
      </Table>
    );

    expect(container).toMatchSnapshot();
  });
});
