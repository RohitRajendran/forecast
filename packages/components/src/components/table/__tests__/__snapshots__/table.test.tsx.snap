// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table should render 1`] = `
<div>
  <table
    class="tp-c-gVqKZS tp-c-gVqKZS-ktwPXh-tableLayout-fixed"
  >
    <thead
      class="tp-c-buOltT"
    >
      <th
        class="tp-c-gSLweG"
      >
        File Name
      </th>
      <th
        class="tp-c-gSLweG"
      >
        Date Uploaded
      </th>
      <th
        class="tp-c-gSLweG"
      >
        Imported By
      </th>
      <th
        class="tp-c-gSLweG"
      >
        Import Status
      </th>
    </thead>
    <tbody
      class="tp-c-lesPJm"
    >
      <tr
        class="tp-c-buOltT"
      >
        <td
          class="tp-c-eMaKFH"
        >
          trueplan2.csv
        </td>
        <td
          class="tp-c-eMaKFH"
        >
          May 24, 2022 at 3:23pm
        </td>
        <td
          class="tp-c-eMaKFH"
        >
          Olivia Smith
        </td>
        <td
          class="tp-c-eMaKFH"
        >
          Failed
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
